### PR TITLE
samples: subsys: display: lvgl: rename and move to samples/modules directory

### PR DESCRIPTION
**Pull request by Ballen7**
Currently LVGL samples exists in `samples/subsys/display/lvgl` and `samples/modules/lvgl`

In this PR I rename the sample `samples/subsys/display/lvgl` to `samples/subsys/display/input_devices` then move it to the `samples/modules/lvgl` directory.

### CODE REVIEW

**Review by faxe1008 — CHANGES_REQUESTED** [review]
Thanks for working on this 👍. A few thoughts on this:

- Should the sample name be `lvgl_input`? It doubles the name since it's already within the `lvgl` folder.
- I think to have this bisectable, squashing some (if not all) commits down would be better
- There are quite a few other mentions of the sample accross the docs:

```
grep -nrI subsys/display/lvgl

samples/subsys/display/lvgl/README.rst:70:   :zephyr-app: samples/subsys/display/lvgl
samples/subsys/display/lvgl/README.rst:78:   :zephyr-app: samples/subsys/display/lvgl
boards/xtensa/heltec_wifi_lora32_v2/doc/index.rst:194:   :zephyr-app: samples/subsys/display/lvgl
boards/shields/buydisplay_2_8_tft_touch_arduino/doc/index.rst:63:   :zephyr-app: samples/subsys/display/lvgl
boards/shields/st7735r/doc/index.rst:56:   :zephyr-app: samples/subsys/display/lvgl
boards/shields/st7789v_generic/doc/index.rst:59:   :zephyr-app: samples/subsys/display/lvgl
boards/shields/waveshare_epaper/doc/index.rst:85:   :zephyr-app: samples/subsys/display/lvgl
boards/shields/ssd1306/doc/index.rst:44:   :zephyr-app: samples/subsys/display/lvgl
boards/shields/buydisplay_3_5_tft_touch_arduino/doc/index.rst:65:   :zephyr-app: samples/subsys/display/lvgl
boards/shields/adafruit_2_8_tft_touch_v2/doc/index.rst:63:   :zephyr-app: samples/subsys/display/lvgl
boards/shields/adafruit_2_8_tft_touch_v2/doc/index.rst:72:   :zephyr-app: samples/subsys/display/lvgl
boards/shields/ls0xx_generic/doc/index.rst:94:   :zephyr-app: samples/subsys/display/lvgl
```

File: samples/modules/lvgl/lvgl_input/README.rst
Comment (kartben) [inline samples/modules/lvgl/lvgl_input/README.rst:1]: as you will be moving an existing sample, please make sure it's added to `doc/_scripts/redirects.py` as we want people browsing e.g. https://docs.zephyrproject.org/3.5.0/samples/subsys/display/lvgl/README.html to be properly redirected when clickin "... The [latest development version](https://docs.zephyrproject.org/latest/samples/subsys/display/lvgl/README.html) of this page [...]"

File: samples/modules/lvgl/lvgl_input/src/main.c
Comment (kartben) [inline samples/modules/lvgl/lvgl_input/src/main.c:1]: Not blocking but rather thinking out loud: this sample has now quite derived from the original / minimalistic hello world, and I think we may need to have both going forward? Maybe the "new" hello world should be literrally just that: a Hello World label in the middle of the screen?

cc @faxe1008

File: samples/modules/lvgl/lvgl_input/README.rst [hunk 1]
```diff
@@ -1,5 +1,5 @@
-.. zephyr:code-sample:: lvgl
-   :name: LVGL basic sample
+.. zephyr:code-sample:: lvgl_input
+   :name: basic LVGL input sample
```
Comment (kartben) [inline samples/modules/lvgl/lvgl_input/README.rst:2]: code sample identifier should be consistent with other LVGL samples + suggested name improvement (also dropping "sample" as it's to be avoided when using the code-sample directive)

```suggestion
.. zephyr:code-sample:: lvgl-input-devices
   :name: LVGL input devices
```

**Comment by kartben** [comment]
> Hi @faxe1008,
> 
> I guess a better name would be `input` or `input_device`? What is your preference? I will clean up the other mentions and also put it all into one commit.
> 
> Thanks

I'd go with "input_devices" (plural) as the folder name, and "lvgl-input-devices" for the `code-sample` identifier

File: boards/xtensa/heltec_wifi_lora32_v2/doc/index.rst [hunk 2]
```diff
@@ -188,10 +188,10 @@ Onboard OLED display
 
 The onboard OLED display is of type ``ssd1306``, has 128*64 pixels and is
 connected via I2C. It can therefore be used by enabling the
-:ref:`ssd1306_128_shield` as shown in the following for the :zephyr:code-sample:`lvgl` sample:
+:ref:`ssd1306_128_shield` as shown in the following for the :zephyr:code-sample:`lvgl_input` sample:
```
Comment (kartben) [inline boards/xtensa/heltec_wifi_lora32_v2/doc/index.rst:191]:
```suggestion
:ref:`ssd1306_128_shield` as shown in the following for the :zephyr:code-sample:`lvgl-input-devices` sample:
```

File: boards/xtensa/heltec_wifi_lora32_v2/doc/index.rst [hunk 3]
```diff
@@ -188,10 +188,10 @@ Onboard OLED display
...
 
 .. zephyr-app-commands::
-   :zephyr-app: samples/subsys/display/lvgl
+   :zephyr-app: samples/modules/lvgl/lvgl_input
```
Comment (kartben) [inline boards/xtensa/heltec_wifi_lora32_v2/doc/index.rst:194]: see other comment re: new folder name

**Review by kartben — CHANGES_REQUESTED** [review]

File: samples/modules/lvgl/lvgl_input/src/main.c
Comment (faxe1008) [reply samples/modules/lvgl/lvgl_input/src/main.c:1]: Yeah one thing that I also thought about was the fact that with the move of this sample there is no sample for the display subsys left, that shows anything with color. Afaik the cfb stuff is limited to mono color? IMO its to be discussed if this should be a lvgl sample or something simpler which just fills the buffer with colors according to the displays color format. 

As for this sample - if you do not have any compatibles other then the pointer one, this is pretty much the hello world sample as it used to be. Sure the code looks more intimidating to someone new to LVGL, if thats your concern.

**Comment by faxe1008** [comment]
> > Hi @faxe1008,
> > I guess a better name would be `input` or `input_device`? What is your preference? I will clean up the other mentions and also put it all into one commit.
> > Thanks
> 
> I'd go with "input_devices" (plural) as the folder name, and "lvgl-input-devices" for the `code-sample` identifier

Yup 👍 for that.

**Review by jfischer-no — CHANGES_REQUESTED** [review]
I do not understand the motivation. The sample has been there for many years, everyone knows where to look. The question is why other samples started to appear in the modules directory.

**Comment by faxe1008** [comment]
> The question is why other samples started to appear in the modules directory.

The reason why the new demo sample was situated there is that it showcases the demos of the lvgl module. I think that its sensible to have the demos for a module inside of `samples/modules/<module>`. Other modules do this as well, eg. the `canopennode` sample does not live inside the `samples/subsys/canbus` folder.
Also, I think the `samples/subsys/display` is a unfortunate place for it since the display is not really a subsystem.

**Comment by kartben** [comment]
I don't agree with the fact that LVGL doesn't belong to samples/modules -- it happens that samples that are there today are only for optional modules, true, but they became optional pretty recently, and samples/modules was deemed a good home for them when they were not optional. 

The "problem" here I guess is more that maybe neither `samples/subsys/display` nor `samples/modules` are perfect homes for something like LVGL, hence why we are having this "former location is not ideal, but it's been like this for a long time so let's keep it", vs. "ya but LVGL is effectively a module :)" conversation.

**Pull request closed without merging**